### PR TITLE
fix(sdk): cloud deployment fails if root directory contains spaces

### PR DESCRIPTION
**Pull request by tsuf239**
fixes: #6465 

terraformAsset can't handle file paths that contain spaces- so I created the bundle zip manually- ONLY if there's a spaced path- since it hurts the benchmarks a bit

## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by tsuf239** [comment]
sdk test run: https://github.com/winglang/wing/actions/runs/11124550295

**Comment by tsuf239** [comment]
another workflow run after a small change: https://github.com/winglang/wing/actions/runs/11127000644

File: packages/@winglang/sdk/src/target-tf-aws/function.ts [hunk 1]
```diff
@@ -284,15 +287,29 @@ export class Function extends cloud.Function implements IAwsFunction {
...
+        path: bundle.directory,
+        type: AssetType.ARCHIVE,
+      });
+
+      this.assetPath = asset.path;
+    }
 
     this.bundleHash = bundle.hash;
-    this.assetPath = asset.path;
   }
```
Comment (Chriscbr) [inline packages/@winglang/sdk/src/target-tf-aws/function.ts:313]: Is there a reason we need to still use TerraformAsset? Can we bundle the zip manually in all cases?

**Review by Chriscbr — APPROVED** [review]

File: packages/@winglang/sdk/src/target-tf-aws/function.ts (hunk 1 above)
Comment (tsuf239) [reply packages/@winglang/sdk/src/target-tf-aws/function.ts:313]: The manual zip makes the process longer in more than 25%- so the benchmark tests are failing (this is why I split it), I think we can consider remove it in the future

**Pull request closed without merging**